agent/src/agent/data_extractor/cacti/cacti.py

```diff
@@ -48,35 +60,83 @@ def extract_metrics(pipeline_: Pipeline, start: str, end: str, step: str) -> lis
             if result[0][2] != int(step):
                 continue
 
-            first_data_item_timestamp = result[0][0]
-            for name_idx, measurement_name in enumerate(result[1]):
-                if measurement_name != item['data_source_name']:
-                    continue
-                for row_idx, data in enumerate(result[2]):
-                    timestamp = int(first_data_item_timestamp) + row_idx * int(step)
-                    value = data[name_idx]
-
-                    # rrd might return a record for the timestamp earlier then start
-                    if timestamp < int(start):
-                        continue
-                    # skip values with timestamp end in order not to duplicate them
-                    if timestamp >= int(end):
-                        continue
-                    # value will be None if it's not available for the chosen consolidation function or timestamp
-                    if value is None:
-                        continue
-
-                    if should_convert_to_bits and pipeline_.config['convert_bytes_into_bits']:
-                        value *= 8
-
-                    metric = deepcopy(base_metric)
-                    metric['properties']['what'] = measurement_name.replace(".", "_").replace(" ", "_")
-                    metric['value'] = value
-                    metric['timestamp'] = timestamp
-                    metrics.append(metric)
+            graph_metrics += _get_values_for_item(
+                result,
+                item,
+                start,
+                end,
+                step,
+                should_convert_to_bits and pipeline_.config['convert_bytes_into_bits'],
+                base_metric
+            )
+        metrics += graph_metrics
+        metrics += _sum_similar(graph_metrics, metrics_to_sum)
     return metrics
 
 
+def _sum_similar(all_metrics: list, metrics_to_sum: dict) -> list:
+    if not metrics_to_sum:
+        return []
+
+    result = {}
+    for metric in all_metrics:
+        if metric['properties']['what'] not in metrics_to_sum:
+            continue
+        key = metric['properties']['what'] + '_' + str(metric['timestamp'])
+        if key in result:
+            result[key]['value'] += metric['value']
+            continue
+        result[key] = deepcopy(metrics_to_sum[metric['properties']['what']])
+        result[key]['properties']['what'] = metric['properties']['what']
+        result[key]['value'] = metric['value']
+        result[key]['timestamp'] = metric['timestamp']
+
+    return list(result.values())
+
+
+def _get_values_for_item(
```

Comment:
I think the name _build_metrics_for_item() or something like this would be better. I mean to use the word `metrics` instead of `values`